book/specifications/kimchi/template.md

```diff
@@ -1,16 +1,87 @@
+Currently, only the [Generic](#double-generic-gate) and the [Poseidon](#poseidon) gates use it (refer to their own sections to see how). 
+All other gates set their values to $0$.
+
+|  row  |   0   |   1   |   2   |   3   |   4   |   5   |   6   |   7   |   8   |   9   |  10   |  11   |  12   |  13   |  14   |
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
+|   0   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |
+
+**Registers (or Witness)**. Registers are also defined at every row, and are split into two types: the *IO registers* from $0$ to $6$ usually contain input or output of the gates (note that a gate can output a value on the next row as well). 
+I/O registers can be wired to each other (they'll be forced to have the same value), no matter what row they're on (for example, the register at `row:0, col:4` can be wired to the register at `row:80, col:6`). 
+The rest of the registers, $7$ through $14$, are called *advice registers* as they can store values that useful only for the row's active gate. 
+Think of them as intermediary or temporary values needed in the computation when the prover executes a circuit.
+
+|  row  |   0   |   1   |   2   |   3   |   4   |   5   |   6   |   7   |   8   |   9   |  10   |  11   |  12   |  13   |  14   |
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
+|   0   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |   /   |
+
+
+**Wiring (or Permutation)**. For gates to take output of other gates, we use a wiring table to wire registers together. 
+It is defined at every row, but only for the first $7$ registers. 
+Each cell specifies a `(row, column)` tuple that it should be wired to. 
+Note that if three or more registered are wired together, they must form a cycle. 
+For example, if register `(0, 4)` is wired to both registers `(80, 6)` and `(90, 0)` then you would have the following table:
+
+|  row  |    0    |   1   |   2   |   3   |    4     |   5   |    6     |
+| :---: | :-----: | :---: | :---: | :---: | :------: | :---: | :------: |
+|   0   |   0,0   |  0,1  |  0,2  |  0,3  | **80,6** |  0,5  |   0,6    |
+|  ...  |         |       |       |       |          |       |          |
+|  80   |  80,0   | 80,1  | 80,2  | 80,3  |   80,4   | 80,5  | **90,0** |
+|  ...  |         |       |       |       |          |       |          |
+|  90   | **0,4** | 90,1  | 90,2  | 90,3  |   90,4   | 90,5  |   90,6   |
+
+**Wiring (Permutation) trace**. You can think of the permutation trace as an extra register that is used to enforce the wiring specified in the wiring table. 
+It is a single column that applies on all the rows as well, which the prover computes as part of a proof.
```

Comment:
Could you explain more about how this is represented and integrated, maybe by giving the structure of the polynomial or formula as it relates to this?

Reply:
This should be specified in the permutation section. But to answer the question quickly, it is supposed to be this polynomial (without the division by the vanishing polynomial, and with 7 columns instead of 3, and [with some part removed that the verifier can compute themselves](https://o1-labs.github.io/mina-book/crypto/plonk/final_check.html))

<img width="1029" alt="Screen Shot 2022-03-11 at 11 48 42 AM" src="https://user-images.githubusercontent.com/1316043/157948102-391e798c-ed7f-4311-87be-45a799f03380.png">

taken from https://o1-labs.github.io/mina-book/crypto/plonk/zkpm.html#zero-knowledge-for-the-permutation-polynomial

Reply:
@querolita the sigma permutation polynomials is what I call "wiring (or permutation)" right above that. Probably a good idea to also call it sigmas.

Reply:
Oooooh I see ok ok... I was thinking of the above polynomial as a separate thing, but never as a possible additional column in the trace.
Regarding the sigmas, thanks for clarifying. These two things are quite interconnected and the way you have split it in two makes it easier.